src/common/runtime/cmdline.ts

```diff
@@ -54,7 +67,33 @@ if (queries.length === 0) {
   try {
     const loader = new DefaultTestFileLoader();
     assert(queries.length === 1, 'currently, there must be exactly one query on the cmd line');
-    const testcases = await loader.loadCases(parseQuery(queries[0]));
+    const testcaseQuery = parseQuery(queries[0]);
+    const testcases = await loader.loadCases(testcaseQuery);
+
+    const expectations: TestQueryWithExpectation[] = [];
+    if (typeof loadWebGPUExpectations !== 'undefined') {
+      (await loadWebGPUExpectations).forEach(entry => {
```

Comment:
I think almost all of this can be factored out, it's almost the same across runtimes

Reply:
Done.